hibernate-core/src/main/java/org/hibernate/query/CommonQueryContract.java

```diff
@@ -20,6 +28,424 @@
  * @author Gavin King
  */
 public interface CommonQueryContract {
```

Comment:
I get why it's necessary to have a common interface and then override all methods in sub-interfaces (to return the sub-interface instead of the common interface).

However, as you know, there's a real risk of forgetting to add overrides when we add a new method to the common interface. Well, it's not even a risk, it's a certainty: we will forget at some point.

On projects using Maven, I would have used [JQAssistant](https://jqassistant.org/) to check that all sub-interfaces of `CommonQueryContract` override all methods that return a `CommonQueryContract` with the proper return type.
But with Gradle, you'd have to use [this unofficial plugin](https://github.com/kontext-e/jqassistant-gradle-plugin), and I'm not sure what it's worth. Plus, setting this up may be complicated.

Still, using this inheritance strategy without automated checks makes me uneasy...